src/features/content-answers/ContentAnswersModal.tsx

```diff
@@ -48,15 +49,17 @@ const ContentAnswersModal = ({ api, currentUser, file, isOpen, onRequestClose }:
         if (lastQuestion && lastQuestion.prompt) {
             lastQuestion.answer = response.data.answer;
             lastQuestion.createdAt = response.data.created_at;
+            setQuestions([...updatedQuestions, lastQuestion]);
         }
-
-        setQuestions([...updatedQuestions, lastQuestion]);
     }, []);
 
-    const handleErrorCallback = useCallback((error: ElementsXhrError): void => {
+    const handleErrorCallback = useCallback((error: ElementsXhrError, updatedQuestions: QuestionType[]): void => {
         setIsLoading(false);
-        if (error) {
-            // TODO: render error component
+
+        const lastQuestion = updatedQuestions.pop();
```

Comment:
we should probably avoid object mutation here since the reference to `updatedQuestions` is used in multiple places which could cause unexpected issues

maybe a `slice()` or length solution:
```
const [lastQuestion] = updatedQuestions.slice(-1);

// or

const lastQuestion = updatedQuestions[updatedQuestions.length - 1];
```